### PR TITLE
fix: add random_state=42 to train_test_split calls for reproducible notebook results

### LINKED ISSUE #1407: Fix: Add random_state to train_test_split calls for reproducible ML notebook results
## Description

Several machine learning notebooks in the repository use train_test_split() without specifying a random_state parameter.

Because of this, the train-test split changes every time the notebook is executed, which causes:

- inconsistent accuracy scores,
- non-reproducible outputs,
- unstable model evaluation,
- and confusion for beginners following the notebooks.

For an educational ML repository, reproducibility is extremely important so that learners can consistently obtain the same results while studying or experimenting with the notebooks.

### Current Behavior

- Example of the current implementation:

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)

Since no random_state is provided, the dataset split changes on every execution.

### Proposed Solution

Add a fixed random_state value (random_state=42) to all affected train_test_split() calls.

- Updated implementation:

X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42
)
### Notebooks Targeted

The fix will be applied to the affected notebooks where reproducibility is currently missing, including notebooks related to:

- Heart Disease Prediction
- Diabetes ML notebooks
- Flood Prediction
- Language Detection
- Startup Profit Prediction
- SVM examples
- GridSearchCV examples

and other confirmed notebooks with missing random_state.

### Why This Fix Is Important

Adding random_state improves:

- reproducibility of experiments
- consistency of notebook outputs
- debugging reliability
- educational clarity for beginners
- overall ML workflow stability

This is a standard best practice in machine learning workflows and educational repositories.

## Additional Notes

Only reproducibility-related changes will be made.
No dataset, model architecture, or training logic will be modified.
Notebook outputs will be cleared before submission to keep the PR clean.

I would like to work on this issue under GSSoC 2026 open source contribution.

### PR DESCRIPTION
**Pull request by Primavat**
## Summary

Fixes #1407 

This PR improves reproducibility across multiple machine learning notebooks by adding `random_state=42` to `train_test_split()` calls where it was previously missing.

Without a fixed random state, every notebook execution generates a different train-test split, leading to inconsistent accuracy scores and non-reproducible outputs. This can be confusing for learners using these notebooks for educational purposes.

---

## What Was Changed

Added `random_state=42` to affected `train_test_split()` calls across multiple notebooks, including notebooks related to:

* Heart Disease Prediction
* Diabetes ML examples
* Flood Prediction
* Language Detection
* Startup Profit Prediction
* SVM examples
* GridSearchCV examples

Example fix:

### Before

```python
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)
```

### After

```python
X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42
)
```

---

## Why This Change Is Necessary

This repository contains educational ML notebooks, where reproducibility and consistency are important for learners and contributors.

Adding a fixed random seed helps ensure:

* consistent model evaluation,
* stable accuracy scores,
* reproducible experiments,
* easier debugging,
* and a better learning experience.

This follows standard machine learning best practices.

---

## Additional Notes

* No model architecture or dataset logic was modified.
* Only reproducibility-related improvements were made.
* Notebook outputs were cleared before commit to keep the PR clean.

## Testing

* Verified that notebooks now generate consistent train-test splits across repeated runs.
* Confirmed that no unintended notebook logic changes were introduced.

### CODE REVIEW

**Comment by Primavat** [comment]
Hey @Niketkumardheeryan,

The large `+/-` diff count in this PR is primarily due to Jupyter notebook (`.ipynb`) formatting and output cleanup, not because of major logic changes.

The actual code changes are limited to adding `random_state=42` to affected `train_test_split()` calls for reproducibility.

Additional notebook-related changes include:

* clearing notebook outputs,
* resetting execution counts,
* automatic JSON formatting/metadata normalization performed by Jupyter.

No ML model logic, dataset handling, or notebook workflows were modified beyond the reproducibility fix.

Core functional change:

* Added `random_state=42` to the affected notebook train/test split calls to ensure deterministic and reproducible results for learners.

Kindly review my PR for approval.

Thank you.

**Comment by Primavat** [comment]
@Niketkumardheeryan kindly review my pr

File: Basics of ML and DL/ML/SVM/SVM-Iris.ipynb
Comment (Sibam-Paul) [inline Basics of ML and DL/ML/SVM/SVM-Iris.ipynb:1]: run the cells

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
please run all the cells

**Comment by Sibam-Paul** [comment]
@Primavat  are u active will close the pr no activity

**Comment by Sibam-Paul** [comment]
@Primavat  are u active will close the pr no activity 
fix the  conflicts

**Comment by Primavat** [comment]
I am not longer active, you may close the pr.
Thank you
